tests/v2/validation/provisioning/provisioning_rke1_driver_test.go

```diff
@@ -0,0 +1,185 @@
+			for _, cni := range r.cnis {
+				name += " cni: " + cni
+				r.Run(name, func() {
+					cluster := &management.Cluster{
+						DockerRootDir:           "/var/lib/docker",
+						EnableClusterAlerting:   false,
+						EnableClusterMonitoring: false,
+						LocalClusterAuthEndpoint: &management.LocalClusterAuthEndpoint{
+							Enabled: true,
+						},
+						Name:     AppendRandomString("rke1"),
+						Provider: provider.Name,
+						RancherKubernetesEngineConfig: &management.RancherKubernetesEngineConfig{
+							DNS: &management.DNSConfig{
+								Provider: "coredns",
+								Options: map[string]string{
+									"stubDomains": "cluster.local",
+								},
+							},
+							Ingress: &management.IngressConfig{
+								Provider: "nginx",
+							},
+							Monitoring: &management.MonitoringConfig{
+								Provider: "metrics-server",
+							},
+							Network: &management.NetworkConfig{
+								MTU:     0,
+								Options: map[string]string{},
+							},
+						},
+					}
+
+					clusterResp, err := client.Management.Cluster.Create(cluster)
+					require.NoError(r.T(), err)
+
+					var nodeTemplateResp *nodetemplates.NodeTemplate
+
+					switch {
+					case provider.Name == "azure":
+						nodeTemplateResp, err = azure.CreateAzureNodeTemplate(client)
+						require.NoError(r.T(), err)
+					case provider.Name == "aws":
+						nodeTemplateResp, err = aws.CreateAWSNodeTemplate(client)
+						require.NoError(r.T(), err)
+					case provider.Name == "linode":
+						nodeTemplateResp, err = linode.CreateLinodeNodeTemplate(client)
+						require.NoError(r.T(), err)
+					default:
+						fmt.Println("No nodetemplate for provider: " + provider.Name)
+					}
+
+					nodePool := &management.NodePool{
+						ClusterID:               clusterResp.ID,
+						ControlPlane:            true,
+						DeleteNotReadyAfterSecs: 0,
+						Etcd:                    true,
+						HostnamePrefix:          AppendRandomString("rke1"),
+						NodeTemplateID:          nodeTemplateResp.ID,
+						Quantity:                1,
+						Worker:                  true,
+					}
+
+					nodePoolResp, err := client.Management.NodePool.Create(nodePool)
+					require.NoError(r.T(), err)
+					fmt.Printf(nodePoolResp.ClusterID)
+
+					clusterName := clusterResp.Name
+
+					opts := metav1.ListOptions{
+						FieldSelector:  "metadata.name=" + clusterResp.ID,
+						TimeoutSeconds: &defaults.WatchTimeoutSeconds,
+					}
+					watchInterface, err := r.client.GetManagementWatchInterface(management.ClusterType, opts)
+					require.NoError(r.T(), err)
+
+					checkFunc := clusters.IsHostedProvisioningClusterReady
+
+					err = wait.WatchWait(watchInterface, checkFunc)
+					require.NoError(r.T(), err)
+					assert.Equal(r.T(), clusterName, clusterResp.Name)
+				})
+			}
+		}
+	}
+}
+
+func (r *RKE1NodeDriverProvisioningTestSuite) TestProvisioning() {
+	for _, providerName := range r.providers {
+		provider := CreateProvider(providerName)
```

Comment:
We need a new CreateProvider specific to RKE1, this is setting up the rke-machine-config.cattle.io which is specific to RKE2

Reply:
Sounds good, I will look into that along with the other review that you gave.